### PR TITLE
GlusterFS refactoring

### PR DESCRIPTION
**Pull request by LLParse**
optimize probe logic by performing leader election, tcp health checks before probing
expose daemon logs to container stdout/stderr
set retain_ip=true for daemon containers
use busybox, add /var/lib/glusterd, remove network interface for data containers

### CODE REVIEW

File: glusterfs/0.2.1/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+glusterfs-server:
+  image: llparse/glusterfs:v0.2.1
```
Comment (cloudnautique) [inline glusterfs/0.2.1/docker-compose.yml]: change back to rancher

File: glusterfs/containers/0.2.1/glusterfs/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+ADD ./*.sh /opt/rancher/
+
+EXPOSE 24007
+EXPOSE 24007/udp
+EXPOSE 24008
+EXPOSE 24008/udp
+EXPOSE 49152
+
+# Add Giddyup v0.8.0
+ADD giddyup /usr/bin/giddyup
```
Comment (cloudnautique) [inline glusterfs/containers/0.2.1/glusterfs/Dockerfile]: can we try going back to GH

File: glusterfs/0.2.1/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    io.rancher.sidekicks: glusterfs-probe,glusterfs-data,glusterfs-volume-create
+  command: glusterd -p /var/run/gluster.pid --no-daemon --log-file=-
+glusterfs-probe:
+  image: llparse/glusterfs:v0.2.1
+  net: ${network_mode}
+  volumes_from:
+    - glusterfs-data
+  labels:
+    io.rancher.container.hostname_override: container_name
+  command: /opt/rancher/peerprobe.sh
```
Comment (cloudnautique) [inline glusterfs/0.2.1/docker-compose.yml]: I thought we were adding a health check and giddyup was going ot be the entrypoint?

File: glusterfs/containers/0.2.1/glusterfs/Dockerfile (hunk 2 above)
Comment (LLParse) [inline glusterfs/containers/0.2.1/glusterfs/Dockerfile]: Yes i'll revert this now that GH is back up

File: glusterfs/0.2.1/docker-compose.yml (hunk 3 above)
Comment (LLParse) [inline glusterfs/0.2.1/docker-compose.yml]: Yes I'll get that change in

**Comment by LLParse** [comment]
Squashed into new branch.
Opened PR https://github.com/rancher/compose-templates/pull/49

**Pull request closed without merging**